server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -741,8 +750,21 @@ public Query multiPhraseQuery(TokenStream stream, int slop, boolean enablePositi
             return createPhraseQuery(stream, field, slop, enablePositionIncrements);
         }
 
+        private int countTokens(TokenStream ts) throws IOException {
+            ts.reset();
```

Comment:
maybe assert that this is a cached tokenstream?

Reply:
this feels a bit restrictive, as it could be a CachedTokenStream wrapped in a filter?